src/algorithms/nspso.cpp

```diff
@@ -141,12 +141,10 @@ population nspso::evolve(population pop) const
         maxv[i] = vwidth;
     }
     // Initialize the particle velocities if necessary
-    if ((m_velocity.size() != swarm_size)) { // if with memory, add: || (!m_memory)
-        m_velocity = std::vector<vector_double>(swarm_size, dummy_vel);
-        for (decltype(swarm_size) i = 0u; i < swarm_size; ++i) {
-            for (decltype(n_x) j = 0u; j < n_x; ++j) {
-                m_velocity[i][j] = uniform_real_from_range(minv[j], maxv[j], m_e);
-            }
+    m_velocity = std::vector<vector_double>(swarm_size, dummy_vel);
```

Comment:
Just for clarity.. this (I believe) was causing the issue.. the line: `(m_velocity.size() != swarm_size` should always be false in my tests, but for some test cases it resulted to be true, thus causing the `m_velocity` not to be re-initialized for some cases, and thus it resulted to have the wrong size.. tomorrow I will investigate better why this happened (if I confirm that this is indeed the issue)

Reply:
Is this coming from the original nspso implementation?

Reply:
As far as I understood, in the previous pagmo the velocity was generated random in the first individual (similar to the decision vector) but in this pagmo is not possible anymore, so I have decided to do the same as it is done in pso.hpp to initialize the velocity

Reply:
In any case, I will have to go through this part again anyway when I will introduce memory